src/parquet_bridge.rs

```diff
@@ -90,12 +90,45 @@ impl From<Compression> for CompressionCodec {
             Compression::Lzo => CompressionCodec::LZO,
             Compression::Brotli => CompressionCodec::BROTLI,
             Compression::Lz4 => CompressionCodec::LZ4,
-            Compression::Zstd => CompressionCodec::ZSTD,
+            Compression::Zstd(_) => CompressionCodec::ZSTD,
             Compression::Lz4Raw => CompressionCodec::LZ4_RAW,
         }
     }
 }
 
+/// Represents a valid zstd compression level.
+#[derive(Debug, Eq, PartialEq, Hash, Clone, Copy)]
+pub struct ZstdLevel(i32);
+
+impl ZstdLevel {
+    /// Attempts to create a zstd compression level from a given compression level.
+    ///
+    /// Compression levels must be valid (i.e. be acceptable for [`zstd::compression_level_range`])
+    pub fn try_new(level: i32) -> Result<Self, Error> {
+        let compression_range = zstd::compression_level_range();
```

Comment:
`zstd` is an optional feature. This forces everyone to have it installed by default, right?

We need to feature-flag this.

Reply:
Yes, that's true, I'll also feature flag the `Default` impl as it pulls the default compression level from `zstd` too.

Reply:
One option for the default is to use `None` and only concretise it to the default in the  
`compression` function, which is already feature flag.

This is also easier to follow when printing it in debug mode - the debug of the metadata in reading is `(None)` instead of `(Some(...))`. The latter is not easy to follow why there is a value there when we have no such information about that (and we won't be using it)

Reply:
Yes good point, I'll get this updated today

Reply:
The feature flags have been added